### PR TITLE
Update upgrade guide

### PR DESCRIPTION
**Pull request by theofidry**

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -1,26 +1,71 @@
...
+
+- Removing the persistence layer. This work should be handled in Alice extensions
+  such as [AliceDataFixtures](https://github.com/theofidry/AliceDataFixtures)
+- Introduce a proper fixture lifecycle (building, instantiation,
+  hydration and configuration step)
+- Introduce a proper Lexer
+
+The aim of those changes are to make Alice more robust, performant and
+extensible. The technical debt and a lot of weird edge cases were extremely hard
+to in 2.x due to its design.
```
Comment (stof) [inline UPGRADE.md]: a verb seems to be missing here after ``to``

File: UPGRADE.md [hunk 2]
```diff
@@ -1,26 +1,71 @@
...
+If you are maintaining an extension, bundle or module for Alice, chances are
+high that you have to redo a big chunk of work. However if you are just a
+simple consumer, little has changed:
+
+- The loading is now done via the new loader `Nelmio\Alice\Loader\NativeLoader`
+- You can easily injected parameters and objects to this loader
+- The loader now returns a set containing both the parameters resolved and the
+  objects created (in 2.x, only the objects were returned)
+- Frameworks bridges are integrated to Alice, so if you are using Symfony for
+  example the laoder is accessible via the `nelmio_alice.data_loader` or
```
Comment (stof) [inline UPGRADE.md]: typo: ``loader``

File: UPGRADE.md [hunk 3]
```diff
@@ -1,26 +1,71 @@
...
+where the result could not be guaranteed or syntax errors. Whenever possible,
+a deprecation message about those changes will land in the patch versions of
+Alice 2.x.
+
+Alice 2.x is no longer supported. Some bug fixes or improvements may still be
+done depending of the case, but no Alice maintainer will actively work on it
+(PR may still be welcomed though).
+
+
 # Breaking changes between Alice 1.x and 2.0
```
Comment (stof) [inline UPGRADE.md]: in the 3.x codebase, you could even drop these instruction entirely

Comment (theofidry) [inline UPGRADE.md]: true

**Pull request merged**